File: src/Core/Tests/RazorEngine.Core.VB.Tests/TestTypes/Activation/UnityTemplateActivator.vb
Imports Microsoft.Practices.Unity
Imports RazorEngine
Imports RazorEngine.Templating



Namespace TestTypes.Activation



    ''' <summary>
    ''' Defines an activator that supports Unity.
    ''' </summary>
    Public Class UnityTemplateActivator
        Implements IActivator
#Region "Fields"
        Private ReadOnly _container As UnityContainer
#End Region

#Region "Constructor"
        ''' <summary>
        ''' Initialises a new instance of <see cref="UnityTemplateActivator"/>.
        ''' </summary>
        ''' <param name="container">The unity container.</param>
        Public Sub New(container As UnityContainer)
            If container Is Nothing Then
                Throw New ArgumentNullException("container")
            End If

            _container = container
        End Sub
#End Region

#Region "Methods"
        ''' <summary>
        ''' Creates an instance of the specifed template.
        ''' </summary>
        ''' <param name="context">The instance context.</param>
        ''' <returns>An instance of <see cref="ITemplate"/>.</returns>
        Public Function CreateInstance(context As InstanceContext) As ITemplate Implements IActivator.CreateInstance
            Return DirectCast(_container.Resolve(context.TemplateType), ITemplate)
        End Function
#End Region
    End Class
End Namespace

File: src/Core/Tests/RazorEngine.Core.Tests/TestTypes/Activation/UnityTemplateActivator.cs
namespace RazorEngine.Tests.TestTypes.Activation
{
    using System;

    using Microsoft.Practices.Unity;

    using Templating;

    /// <summary>
    /// Defines an activator that supports Unity.
    /// </summary>
    public class UnityTemplateActivator : IActivator
    {
        #region Fields
        private readonly UnityContainer _container;
        #endregion

        #region Constructor
        /// <summary>
        /// Initialises a new instance of <see cref="UnityTemplateActivator"/>.
        /// </summary>
        /// <param name="container">The unity container.</param>
        public UnityTemplateActivator(UnityContainer container)
        {
            if (container == null)
                throw new ArgumentNullException("container");

            _container = container;
        }
        #endregion

        #region Methods
        /// <summary>
        /// Creates an instance of the specifed template.
        /// </summary>
        /// <param name="context">The instance context.</param>
        /// <returns>An instance of <see cref="ITemplate"/>.</returns>
        public ITemplate CreateInstance(InstanceContext context)
        {
            return (ITemplate)_container.Resolve(context.TemplateType);
        }
        #endregion
    }
}